db/python/tables/project.py

```diff
@@ -120,6 +120,16 @@ async def get_projects_accessible_by_user(
 
         return project_id_map, project_name_map
 
+    async def get_project_by_id(self, project_id: int) -> Project:
```

Comment:
I'm not sure we need this for this change specifically (see other comments), but it would be handy to have a get project by id method. But if we are adding it, I think it would be better suited on the connection which already has a map of projects and similar methods for getting projects by ids. 

https://github.com/populationgenomics/metamist/blob/dev/db/python/connect.py#L143